### DISCUSSION TITLE
Drone is not stable even with Flow deck attached

### DISCUSSION THREAD

**Discussion by Jimegroxak** [bitcraze/discussions#334] [Q&A]
Hey everyone,

I'm currently working as a research assistant doing work with a couple of crazyflie 2.0 drones and an LPS positioning system. Only problem is I am unable to get any of the to even hover with any stability. Every drone ends up veering away and crashing into a wall for whatever reason. So far we have tried just taking off and landing with cfclient, as well as trying to use code to do the same behavior through three different commanders: motion, position, and high-level, with no success. We have been testing the propellers and battery and they all come back fine. We have tried the drones without the flow deck and with several different decks as well. The firmware is updated to the latest version. Specific version numbers will be attached below.

Drones:
Crazyflie 2.0 w/ flowdeck and lps deck attached
version 2022.01 and 2022.09

LPS:
8 LPS anchors in TDoA 3 mode

CFclient 2021.6.1.post84+b6c5eb2
Ubuntu 20.04.3 LTS

If anyone has any advice or has encountered a similar problem before please let me know!

**Comment by Jimegroxak** [comment]
Update: the drones seem to fly well when I remove the lps deck, so I'm assuming that's where the issue is.

**Reply by knmcguire** [reply]
Could you connect to the cfclient and give us the console output with both decks attached? Perhaps there is something going on since they should be compatible.

**Reply by knmcguire** [reply]
also, what kind of surface are you flying over? Black, shiny, low texture?

**Reply by Jimegroxak** [reply]
Here's the console output.

SYS: ----------------------------
SYS: Crazyflie 2.0 is up and running!
SYS: Build 112:9cba816743df (2022.01 +112) CLEAN
SYS: I am 0x303436313037471300370028 and I have 1024KB of flash!
CFGBLK: v1, verification [OK]
DECK_CORE: 2 deck(s) found
DECK_CORE: Calling INIT on driver bcFlow for deck 0
PMW: Motion chip id: 0x49:0xB6
DECK_CORE: Calling INIT on driver bcDWM1000 for deck 1
MPU9250 I2C connection [OK].
AK8963 I2C connection [OK].
LPS25H I2C connection [OK].
ESTIMATOR: Using Kalman (2) estimator
CONTROLLER: Using PID (1) controller
MTR-DRV: Using brushed motor driver
PLATFORM: Low interference mode. NRF51 TX power offset by -12db.
SYS: About to run tests in system.c.
EEPROM: I2C connection [OK].
STORAGE: Storage check [OK].
AK8963: Self test [OK].
DECK_CORE: Deck 0 test [OK].
DECK_CORE: Deck 1 test [OK].
SYS: Self test passed!
STAB: Wait for sensor calibration...
SYS: Free heap: 15000 bytes
STAB: Ready to fly.
DWM: Automatic mode: detected TDoA3
STAB: WARNING: stabilizer loop rate is off (906)

And we're flying it over a rug that should be textured enough for the flowdeck to work.

**Reply by krichardsson** [reply]
> the drones seem to fly well when I remove the lps deck, so I'm assuming that's where the issue is.

That is indicating that the Flow deck is OK. Good!

Is the Loco system working OK? "Bad" position data from the Loco system would also cause instabilities. What happens if you have the Loco deck mounted but turns the anchors off?
